src/api/api.cpp

```diff
@@ -40,24 +40,29 @@
 #include <migraphx/convert_to_json.hpp>
 #include <algorithm>
 #include <cstdarg>
-namespace migraphx {
 
+#ifdef MIGRAPHX_BUILD_TESTING
 static thread_local bool disable_exception_catch = false; // NOLINT
 
 extern "C" MIGRAPHX_C_EXPORT void migraphx_test_private_disable_exception_catch(bool b)
 {
     disable_exception_catch = b;
 }
+#endif
```

Comment:
This should be in the `migraphx` namespace.

Reply:
The purpose of this PR is to move the entire function out of the `migraphx` namespace.
On Windows, putting the function in the `migraphx` namespace pollutes the API of the `migraphx_c` library (which is supposed to be a pure C API library) with C++ mangled symbols. This results in issues while linking `migraphx_c` to C applications (i.e., `migraphx_c` test) because of the missing `ubsan` library. The `ubsan` library is a Clang undefined behavior sanitizer only available for C++ applications and not for C applications.

Reply:
> On Windows, putting the function in the migraphx namespace pollutes the API of the migraphx_c library (which is supposed to be a pure C API library) with C++ mangled symbols

We use `extern "C"` on the function so the name shouldn't be mangled. 

> This results in issues while linking migraphx_c to C applications (i.e., migraphx_c test) because of the missing ubsan library. The ubsan library is a Clang undefined behavior sanitizer, and it seems, on Windows, it is only available for C++ applications and not C applications.

I am not sure how this is related, or how this would even fix that issue.